File: PATTERNS/Pattern46/Pattern46.go
package main

import "fmt"

func main() {
	var rows int
	fmt.Print("Enter the number of rows: ")
	fmt.Scan(&rows) // Number of rows in the butterfly pattern

	// Print the upper half of the butterfly pattern
	for i := 1; i <= rows; i++ {
		// Print numbers in increasing order
		for j := 1; j <= i; j++ {
			fmt.Printf("%d ", j)
		}

		// Print spaces
		for j := 1; j <= 2*(rows-i); j++ {
			fmt.Print("  ")
		}

		// Print numbers in decreasing order
		for j := i; j >= 1; j-- {
			fmt.Printf("%d ", j)
		}

		fmt.Println()
	}

	// Print the lower half of the butterfly pattern
	for i := 1; i <= rows; i++ {
		// Print spaces
		for j := 1; j <= i; j++ {
			fmt.Print("  ")
		}

		// Print numbers in decreasing order
		for j := rows; j >= i; j-- {
			fmt.Printf("%d ", j)
		}

		// Print numbers in decreasing order
		for j := rows - i; j >= 1; j-- {
			fmt.Printf("%d ", j)
		}

		fmt.Println()
	}
}

File: PATTERNS/Pattern46/Pattern46.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('Enter the number of rows: ', (rows) => {
  // Convert rows to a number
  rows = parseInt(rows);

  // Print the upper half of the butterfly pattern
  for (let i = 1; i <= rows; i++) {
    // Print numbers in increasing order
    for (let j = 1; j <= i; j++) {
      process.stdout.write(j + ' ');
    }

    // Print spaces
    for (let j = 1; j <= 2 * (rows - i); j++) {
      process.stdout.write('  ');
    }

    // Print numbers in decreasing order
    for (let j = i; j >= 1; j--) {
      process.stdout.write(j + ' ');
    }

    console.log();
  }

  // Print the lower half of the butterfly pattern
  for (let i = 1; i <= rows; i++) {
    // Print spaces
    for (let j = 1; j <= i; j++) {
      process.stdout.write('  ');
    }

    // Print numbers in decreasing order
    for (let j = rows; j >= i; j--) {
      process.stdout.write(j + ' ');
    }

    // Print numbers in decreasing order
    for (let j = rows - i; j >= 1; j--) {
      process.stdout.write(j + ' ');
    }

    console.log();
  }

  rl.close();
});
